hadoop-ozone/integration-test-s3/src/test/java/org/apache/hadoop/ozone/s3/awssdk/v2/AbstractS3SDKV2Tests.java

```diff
@@ -383,6 +388,215 @@ public void testCopyObject() {
   }
 
+  @Test
+  public void testPutAndGetObjectTagging() {
+    final String bucketName = getBucketName();
+    final String keyName = getKeyName();
+    final String content = "test content";
+    s3Client.createBucket(b -> b.bucket(bucketName));
+
+    s3Client.putObject(b -> b
+            .bucket(bucketName)
+            .key(keyName),
+        RequestBody.fromString(content));
+
+    List<Tag> tags = Arrays.asList(
+        Tag.builder().key("env").value("test").build(),
+        Tag.builder().key("project").value("ozone").build()
+    );
+
+    s3Client.putObjectTagging(b -> b
+        .bucket(bucketName)
+        .key(keyName)
+        .tagging(Tagging.builder().tagSet(tags).build()));
+
+    GetObjectTaggingResponse getResponse = s3Client.getObjectTagging(b -> b
+        .bucket(bucketName)
+        .key(keyName));
+
+    assertEquals(tags.size(), getResponse.tagSet().size());
+    Map<String, String> tagMap = getResponse.tagSet().stream()
+        .collect(Collectors.toMap(Tag::key, Tag::value));
+    assertEquals("test", tagMap.get("env"));
+    assertEquals("ozone", tagMap.get("project"));
+  }
+
+  @Test
+  public void testDeleteObjectTagging() {
+    final String bucketName = getBucketName();
+    final String keyName = getKeyName();
+    final String content = "test content";
+    s3Client.createBucket(b -> b.bucket(bucketName));
+
+    s3Client.putObject(b -> b
+            .bucket(bucketName)
+            .key(keyName),
+        RequestBody.fromString(content));
+
+    List<Tag> tags = Arrays.asList(
+        Tag.builder().key("temp").value("data").build()
+    );
+
+    s3Client.putObjectTagging(b -> b
+        .bucket(bucketName)
+        .key(keyName)
+        .tagging(Tagging.builder().tagSet(tags).build()));
+
+    GetObjectTaggingResponse beforeDelete = s3Client.getObjectTagging(b -> b
+        .bucket(bucketName)
+        .key(keyName));
+    assertEquals(1, beforeDelete.tagSet().size());
+
+    s3Client.deleteObjectTagging(b -> b
+        .bucket(bucketName)
+        .key(keyName));
+
+    GetObjectTaggingResponse afterDelete = s3Client.getObjectTagging(b -> b
+        .bucket(bucketName)
+        .key(keyName));
+    assertTrue(afterDelete.tagSet().isEmpty());
+  }
+
+  @Test
+  public void testPutObjectTaggingExceedsLimit() {
+    final String bucketName = getBucketName();
+    final String keyName = getKeyName();
+    s3Client.createBucket(b -> b.bucket(bucketName));
+    s3Client.putObject(b -> b.bucket(bucketName).key(keyName),
+        RequestBody.fromString("content"));
+
+    List<Tag> tags = new ArrayList<>();
+    for (int i = 1; i <= 11; i++) {
+      tags.add(Tag.builder().key("key" + i).value("value" + i).build());
+    }
+
+    S3Exception exception = assertThrows(S3Exception.class, () ->
+        s3Client.putObjectTagging(b -> b
+            .bucket(bucketName)
+            .key(keyName)
+            .tagging(Tagging.builder().tagSet(tags).build())));
+    assertEquals(400, exception.statusCode());
```

Comment:
```suggestion
    assertEquals(HTTP_BAD_REQUEST, exception.statusCode());
```
Use `HTTP_BAD_REQUEST` to replace `400`.